### PR TITLE
[GPU] MaxPool-8

### PR DESCRIPTION
**Pull request by okhovan**
- added MaxPool-8 features: dilations and 2nd output for indices.

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl [hunk 1]
```diff
@@ -91,11 +94,15 @@ KERNEL(pooling_gpu)(
 
     ACCUMULATOR_TYPE result = INIT_VAL;
 
+#ifdef SELECTED_INDICES_TYPE
+    SELECTED_INDICES_TYPE result_idx = 0;
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl:98]: I think we can use `uint `type for this variable

File: model-optimizer/version.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+custom_feature/maxpool8_98b94c4c05df20cfc72b0720709b46c450f55526
```
Comment (sshlyapn) [inline model-optimizer/version.txt:1]: Please reset this change

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl [hunk 3]
```diff
@@ -107,6 +114,16 @@ KERNEL(pooling_gpu)(
     uint num_elementes = 0;
 #endif
 
+#ifndef DILATION_SIZE_X
+    #define DILATION_SIZE_X 1
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl:118]: Please add undefs for your defines

**Review by sshlyapn — APPROVED** [review]
In general looks good to me

File: model-optimizer/version.txt (hunk 2 above)
Comment (okhovan) [reply model-optimizer/version.txt]: Done.

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl (hunk 3 above)
Comment (okhovan) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl:118]: Done.

File: inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl (hunk 1 above)
Comment (okhovan) [reply inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl]: Done.

**Comment by sshlyapn** [comment]
@okhovan there is a bug related to RTTI that causes gpu test failures. Could you please update your PR as Roman did in his PR: https://github.com/openvinotoolkit/openvino/pull/9204?
Please take a look on two modifications in his PR:

- src/core/include/openvino/op/i420_to_rgb.hpp

`BWDCMP_RTTI_DECLARATION;`

- src/core/src/op/i420_to_bgr.cpp

`BWDCMP_RTTI_DEFINITION(ov::op::v8::I420toBGR);`

You need to do something similar for MaxPool operation to solve functional test failures

BTW, please take a look on cldnn_unit tests:
```
fusings_gpu/pooling_scale_activation_quantize.basic/0
fusings_gpu/pooling_scale_activation_quantize.i8_output_data_type/0
fusings_gpu/pooling_scale_activation_quantize.i8_output_data_type/0
```
and some other `fusings_gpu/pooling*` tests - they are failed in CI

**Comment by okhovan** [comment]
> @okhovan there is a bug related to RTTI that causes gpu test failures.:
RTTI definitions added. 

> BTW, please take a look on cldnn_unit tests:
Tests are fixed (at least locally fusings_gpu/pooling* tests are passing now).

**Pull request merged**